### ISSUE TITLE
Okta auth failed: Could not retrieve list of MFA methods

### ISSUE THREAD

**Issue by vijay-gs-GitHub** [dowjones/tokendito#33]
Unable to get the MFA methods when connecting from News Network .  

```
ERROR [okta_helpers.py:user_mfa_challenge():110]: Okta auth failed: Could not retrieve list of MFA methods
DEBUG [okta_helpers.py:user_mfa_challenge():113]: Error parsing response: {"expiresAt": "2020-11-15T23:38:17.000Z", "status": "SUCCESS", "sessionToken": "20111c-6ROAZPBqfTH1xG6TziW9rDALHdXRbhigqY9MEi1T-zcpJTu7", "_embedded": {"user": {"id": "00un6xkphiqxfrQK60x7", "profile": {"login": "vijayakumar.senthilkumar@news.com.au", "firstName": "Vijayakumar", "lastName": "Gobichettipalayam Senthilkumar", "locale": "en", "timeZone": "America/Los_Angeles"}}}, "_links": {"cancel": {"href": "https://newscorp.okta.com/api/v1/authn/cancel", "hints": {"allow": ["POST"]}}}}
```

It's failing at line no 108 (**_okta_helpers.py_**)  .  
`mfa_options = primary_auth['_embedded']['factors'] 
`

There is already a **_sessionToken_** available in the OKTA response which was not handled in the code .  Adding below lines of code in **_okta_helpers.py_**  before the _line no 108_ resolved the issue for me .

```
session_token = primary_auth['sessionToken']
   if session_token is not None:
        return session_token   
```

**Comment by vijay-gs-GitHub** [comment]
@kittysyddy Could you please check this issue

**Comment by dan-newscorp** [comment]
+1 Hi, yeah getting same issue as well 

`ERROR [okta_helpers.py:user_mfa_challenge():110]: Okta auth failed: Could not retrieve list of MFA methods
DEBUG [okta_helpers.py:user_mfa_challenge():113]: Error parsing response: {"expiresAt": "2020-11-17T23:20:39.000Z", "status": "SUCCESS", "sessionToken": "20111Bxs-Q2tlbDU2QsUbUOg1tnXmlUb2MzvmTS11RguyHpERbWKjKS", "_embedded": {"user": {"id": "00umso4mywr31Ec5m0x7", "profile": {"login": "daniel.moore@news.com.au", "firstName": "Daniel", "lastName": "Moore", "locale": "en", "timeZone": "America/Los_Angeles"}}}, "_links": {"cancel": {"href": "https://newscorp.okta.com/api/v1/authn/cancel", "hints": {"allow": ["POST"]}}}}`

**Comment by JJLewis** [comment]
+1 Hi, still getting this issue. Only when I am on the company's internal network though.

**Comment by JJLewis** [comment]
#34 #32 
I think either of those PRs would solve this issue.
Personally, I tried the suggested snippet above, and it worked.

**Comment by pcmxgti** [comment]
@JJLewis @vijay-gs-GitHub this has been addressed on version `1.1.2` (#39), please test if it fixes the issue for you.
